src/test/java/com/artipie/FilesRepoITCase.java

```diff
@@ -130,8 +135,35 @@ void curlPutShouldSaveFile() throws Exception {
         );
     }
 
+    @ParameterizedTest
+    @ValueSource(strings = {"PUT", "GET"})
+    void curlPutAndGetShouldFailWithUnauthorized(final String req) throws Exception {
+        this.init(this.config(false));
+        MatcherAssert.assertThat(
+            this.exec(
+                "curl", "-i", "-X", req, String.format(FilesRepoITCase.URL, this.port),
+                "--user", String.format(
+                    "%s:bad%s", ArtipieServer.ALICE.name(), ArtipieServer.ALICE.password()
+                )
+            ),
+            new StringContains("HTTP/1.1 401 Unauthorized")
+        );
+    }
+
+    @Test
+    void curlPutShouldFailWithForbidden() throws Exception {
+        this.init(this.config(false));
+        MatcherAssert.assertThat(
+            this.exec(
+                "curl", "-i", "-X", "PUT", String.format(FilesRepoITCase.URL, this.port),
+                this.flag(false), this.user(false, ArtipieServer.BOB)
+            ),
+            new StringContains("HTTP/1.1 403 Forbidden")
+        );
+    }
```

Comment:
@genryxy please also add a test to see that `download` permission is checked as well. That will require to do `GET` with user who lacks `download` permission